From 2558321ddd1bb2c5b46d6141bd3d1d1009a21061 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Mon, 16 Dec 2019 18:57:36 +0200
Subject: [PATCH 1035/1239] include/configs: use 2MB boot on 9130-CRB mmc
 device

CN9130-CRB has eMMC with 2MB BOOT0/BOOT1 partitions.
This is less than expected (4MB), so the environment location
should be adjusted.

Change-Id: I77d779eca8b6d8d7d3533262bc6bf1fb2179b803
Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/20405
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
---
 include/configs/mvebu_armada-common.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/configs/mvebu_armada-common.h b/include/configs/mvebu_armada-common.h
index 3d3174cb98..c034b657f5 100644
--- a/include/configs/mvebu_armada-common.h
+++ b/include/configs/mvebu_armada-common.h
@@ -116,7 +116,7 @@
 /*
  * Assume minimum flash/eMMC boot partition size of 4MB
  * and save the environment at the end of the boot device
- * There is one exclusion from this rule - the EspressoBIN board with eMMC.
+ * There are some exclusions from this rule, for instance EspressoBIN board.
  * The eMMC device found on some EspressoBIN V7 boards has 2MB boot partition.
  */
 #define CONFIG_ENV_SIZE			(64 << 10) /* 64KiB */
@@ -126,9 +126,11 @@
 #define CONFIG_ENV_OFFSET		0x400000
 #else
 #if defined(CONFIG_ENV_IS_IN_MMC) && \
-defined(CONFIG_TARGET_MVEBU_ARMADA_37XX) && defined(CONFIG_MV88E6XXX_SWITCH)
-/* This one should be EspressoBin, since A3700 DB does not have
- * MV88E6XXX switch enabled. Is there any other way to detect it?
+((defined(CONFIG_TARGET_MVEBU_ARMADA_37XX) && \
+	defined(CONFIG_MV88E6XXX_SWITCH))  || \
+defined(CONFIG_TARGET_OCTEONTX2_CN913x) && (CONFIG_MVEBU_BOOT_PART != 0))
+/* A3700 + MV88E6XXX == EspressoBIN board OR
+ * CN913x CRB that boots from eMMC partition BOOT0/BOOT1 (ID 1/2)
  */
 #define CONFIG_ENV_OFFSET		(0x200000 - CONFIG_ENV_SIZE)
 #else
-- 
2.29.0

